### ISSUE TITLE
BasicMessageHandler only saves the last recipient

### ISSUE THREAD

**Issue by jakobkmar** [davidmoten/subethasmtp#36]
The BasicMessageHandler class does only save the last recipient.

The `recipient()` method can actually be called multiple times (javadoc):
> Called once for every RCPT TO during a SMTP exchange.

Currently, inside of the `recipient()` method the variable "recipient" just gets overridden:
```java
@Override
public void recipient(String recipient) throws RejectException {
    this.recipient = recipient;
}
```

**Comment by davidmoten** [comment]
Righto thanks, I'll have a look.

**Comment by bluefireoly** [comment]
Fixing this will probably be a breaking change, as the type of the `to` parameter will change from `String` to `List<String>`.

**Comment by vodasams57** [comment]
What is the state of this issue? Multiple `RCPT TO` are fundamental features of the SMTP protocol.

**Comment by dcolazin** [comment]
> Fixing this will probably be a breaking change, as the type of the `to` parameter will change from `String` to `List<String>`.

Yes, the `BasicMessageListener` must change, but as seen in the `MessageHandler::recipient` javadoc, the method should be called once for each command. Should I prepare a pull request to handle multiple recipients?

**Comment by davidmoten** [comment]
See this [answer](https://github.com/davidmoten/subethasmtp/issues/144) for a solution. I'm going to close.
